File: src/main/java/co/hisabsoftware/polling/webback/services/mapping/MapperService.java
package co.hisabsoftware.polling.webback.services.mapping;

import co.hisabsoftware.common.utils.StringUtils;
import co.hisabsoftware.polling.webback.Entities.Poll;
import co.hisabsoftware.polling.webback.models.PollDto;
import org.springframework.stereotype.Service;

@Service
public class MapperService {
    public PollDto toDto(Poll poll) {
        return new PollDto(poll.getId(), poll.getTitle());
   }

   public Poll toNewEntity(PollDto pollDto) {
        // TODO : Change DTO to reflect Poll
        return new Poll(pollDto.getId(), pollDto.getText(), StringUtils.EMPTY, true, true);
   }

   public Poll toUpdatedEntity(Poll poll, PollDto pollDto) {
        poll.setId(pollDto.getId());
        poll.setTitle(pollDto.getText());
        // TODO : Set all properties
        return poll;
   }
}

File: src/main/kotlin/co.hisabsoftware.polling.webback/mapping/MapperService.kt
package co.hisabsoftware.polling.webback.mapping

import co.hisabsoftware.common.utils.StringUtils
import co.hisabsoftware.polling.webback.entities.Poll
import co.hisabsoftware.polling.webback.models.PollDto
import org.springframework.stereotype.Service

@Service
class MapperService {
    fun toDto(poll: Poll): PollDto {
        return PollDto(poll.id, poll.title)
    }

    fun toNewEntity(pollDto: PollDto): Poll {
        // TODO : Change DTO to reflect Poll
        return Poll(pollDto.id, pollDto.title, StringUtils.EMPTY, true, true)
    }

    fun toUpdatedEntity(poll: Poll, pollDto: PollDto): Poll {
        poll.id = pollDto.id
        poll.title = pollDto.title
        // TODO : Set all properties
        return poll
    }
}
